### PR TITLE
Added JCL template for RMUTL

### PR DESCRIPTION
**Pull request by Samarinnayak**

### CODE REVIEW

File: galaxy.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ namespace: ibm
 name: ibm_zos_cics
 
 # The collection version
-version: 2.1.0
+version: 2.2.0
```
Comment (andrewhughes101) [inline galaxy.yml:11]: Can we revert this, we do version bumps in dedicated release PR's

File: plugins/module_utils/_data_set_utils.py [hunk 2]
```diff
@@ -273,6 +274,85 @@ def _execute_command(command):
...
     return module.run_command(command)
 
+def _submit_jcl(jcl_uss_path, job_name):
+    executions =[]
+    command = "jsub -f '{0}'".format(jcl_uss_path)
+
+    rc, stdout,stderr = _execute_command(command)
+    executions.append(
+    _execution(
+        name="Submit jcl job {0} for {1}".format(jcl_uss_path, job_name),
```
Comment (andrewhughes101) [inline plugins/module_utils/_data_set_utils.py:284]: Does the file path name make sense here in the first format?
Would read like `Submit jcl job /tmp/file/path for DFHRMUTL`
Also we should capitalize JCL in any externals

File: plugins/module_utils/_global_catalog.py [hunk 3]
```diff
@@ -6,14 +6,14 @@
 # FOR INTERNAL USE IN THE COLLECTION ONLY.
 
 from __future__ import (absolute_import, division, print_function)
+import os
```
Comment (andrewhughes101) [inline plugins/module_utils/_global_catalog.py:9]: Can we not do open imports like this. We only use `os.remove` so dont need to import the entirety of os. Same goes for tempfile below

File: plugins/module_utils/_global_catalog.py [hunk 4]
```diff
@@ -67,64 +55,137 @@ def _get_catalog_records(stdout):  # type: (str) -> tuple[str | None, str | None
...
+        qualified_file_path = fp.name
+    return qualified_file_path
+
+def _validate_line_length(line, name):
+    """
+    Validates that the JCL line does not exceed MAX_LINE_LENGTH.
+    Raises ValueError if validation fails.
+    """
+    if len(line) > MAX_LINE_LENGTH:
+        raise ValueError(f"{name} line exceeds {MAX_LINE_LENGTH} characters: {len(line)}")
```
Comment (andrewhughes101) [inline plugins/module_utils/_global_catalog.py:103]: This is a really good check to do I like it, but the error doesnt make sense to a user. They're gonna see this if the STEPLIB or DFHGCD DSN is too long which would create a length error in the JCL. Instead of telling them they have a JCL error, why dont we tell them exactly which parameter they used was too long, as we should be abstracting that we're even using JCL from them

File: plugins/module_utils/_global_catalog.py [hunk 5]
```diff
@@ -67,64 +55,137 @@ def _get_catalog_records(stdout):  # type: (str) -> tuple[str | None, str | None
...
+    return True
+
+def _validate_name_params(param):
+    if any(len(part) > MAX_NAME_LENGTH for part in param.split('.')):
```
Comment (andrewhughes101) [inline plugins/module_utils/_global_catalog.py:107]: Is this needed? We do upfront checking with better arg parser that both SDFHLOAD and DFHGCD dsn would go through and that would do the 8 char check.

File: plugins/module_utils/_global_catalog.py [hunk 6]
```diff
@@ -67,64 +55,137 @@ def _get_catalog_records(stdout):  # type: (str) -> tuple[str | None, str | None
...
-            reason_code = _get_reason_code(stdout_with_rc)
-            if reason_code and reason_code != "A8":
-                raise MVSExecutionException(
-                    "DFHRMUTL failed with RC 16 - {0}".format(stdout_with_rc[0]), executions
-                )
-            elif reason_code is None:
+    try:
+        for x in range(MVS_CMD_RETRY_ATTEMPTS):
+            dfhrmutl_response, jcl_executions = _execute_dfhrmutl(qualified_file_path)
+            dfhrmutl_rc = dfhrmutl_response.get("ret_code").get("code")
```
Comment (andrewhughes101) [inline plugins/module_utils/_global_catalog.py:129]: We're chaining a `.get` here. Is the first .get because we're not sure if ret_code is always a key in dfhrmutl_response? If its not always a key, that second .get will try and do a .get on a None. If it is always a key we should just use ["ret_code"] instead

File: plugins/module_utils/_global_catalog.py [hunk 7]
```diff
@@ -67,64 +55,137 @@ def _get_catalog_records(stdout):  # type: (str) -> tuple[str | None, str | None
...
 
-    if cmd != "":
-        return executions
+def _execute_dfhrmutl(rmutl_jcl_path, job_name="DFHRMUTL"):
+    executions = _submit_jcl(rmutl_jcl_path, job_name)
+    job_id = executions[0].get("stdout").strip()
 
-    return executions, _get_catalog_records(dfhrmutl_response.stdout)
+    #Give RMUTL a second to run
+    sleep(1)
```
Comment (andrewhughes101) [inline plugins/module_utils/_global_catalog.py:183]: Can we make this an obvious constant instead of just the number here. I know we only use it once but would be good to make it obvious

File: plugins/module_utils/_global_catalog.py [hunk 8]
```diff
@@ -156,3 +217,5 @@ def _get_idcams_cmd_gcd(dataset):   # type: (dict) -> dict
 CI_PERCENT = 10
 CA_PERCENT = 10
 SHARE_CROSSREGION = 2
+MAX_LINE_LENGTH = 72
```
Comment (andrewhughes101) [inline plugins/module_utils/_global_catalog.py:220]: I know the existing constants were here, I'm wondering if we should separate these out a bit as these 2 new ones are not part of the IDCAMS statement unlike all the other. A simple newline and comment might suffice, but also feel free to rearrange them in the file, whichever makes sense

File: tests/unit/helpers/data_set_helper.py [hunk 9]
```diff
@@ -418,6 +419,103 @@ def CSDUP_add_group_stdout(data_set_name):
...
+def get_job_output_execution(rc=0):
+    job = get_sample_job_output()
+    return _execution(
+        name="Get job output for {0}".format(job["job_id"]),
+        rc=rc,
+        stdout=job.get("ret_code").get("msg"),
+        stderr=job.get("ret_code").get("msg_txt")
+    )
+
+def _get_job_dd_output(ddname, job_id):
```
Comment (andrewhughes101) [inline tests/unit/helpers/data_set_helper.py:431]: Why have we got some private methods and non private methods here? The underscores seem inconsistent

File: tests/unit/module_utils/test_dataset_utils.py [hunk 10]
```diff
@@ -700,3 +708,147 @@ def test__write_jcl_to_data_set_fail():
 
     assert e.value.message == "Failed to copy JCL content to data set"
     assert e.value.executions == expected_executions
+
+def test_data_set_utils_submit_jcl():
```
Comment (andrewhughes101) [inline tests/unit/module_utils/test_dataset_utils.py:712]: Dont think the test methods need data_set_utils in them, the files already named accordingly

File: tests/unit/module_utils/test_global_catalog.py [hunk 11]
```diff
@@ -201,20 +203,59 @@ def test_global_catalog_get_records_autocold_emergency():
...
+        assert result == '/mocked/path/to/tempfile'
+
+def test_global_catalog_validate_line_length():
+    line = "//STEPLIB  DD DSNAME=CTS560.CICS730.SDFHLOAD,DISP=SHR"
+    name = "SDFHLOAD"
+    result = global_catalog._validate_line_length(line, name)
+    assert result is True
+
+def test_global_catalog_validate_line_length_failed():
+    line = "//STEPLIB  DD DSNAME=CTS560.CICS730.SDFHLOAD,DISP=SHR this is inavlid line with more the 72 chars"
```
Comment (andrewhughes101) [inline tests/unit/module_utils/test_global_catalog.py:227]:
```suggestion
    line = "//STEPLIB  DD DSNAME=CTS560.CICS730.SDFHLOAD,DISP=SHR this is invalid line with more the 72 chars"
```

File: tests/unit/modules/test_global_catalog.py [hunk 12]
```diff
@@ -102,6 +103,7 @@ def test_create_an_intial_global_catalog():
                 stdout=IDCAMS_create_stdout(NAME),
                 stderr="",
             ),
+            [],
```
Comment (andrewhughes101) [inline tests/unit/modules/test_global_catalog.py:106]: Think you're gonna have to walk me through why we have so many empty arrays in the test executions

**Review by andrewhughes101 — CHANGES_REQUESTED** [review]
Also your sanity test output, I can see a bunch of places where whitespace is gonna get flagged up.

File: plugins/module_utils/_global_catalog.py (hunk 4 above)
Comment (Tom-Latham) [reply plugins/module_utils/_global_catalog.py:103]: Are you thinking something like
`f"Value supplied for {parameter} ({name}) is {len(line)- MAX_LINE_LENGTH} characters too long"`
Where parameter is `cics_region_data_sets` or `region_data_sets`

Comment (andrewhughes101) [reply plugins/module_utils/_global_catalog.py:103]: Yep pretty much, that way its more obvious to a user what they can change to make the error go away

**Comment by Tom-Latham** [comment]
Re-launch CI

**Review by andrewhughes101 — APPROVED** [review]

**Comment by Tom-Latham** [comment]
Re-launch CI
